autotest.pm

```diff
@@ -212,33 +214,31 @@ sub start_process {
     $child->autoflush(1);
     $isotovideo->autoflush(1);
 
-    my $testpid = fork();
-    if ($testpid) {
-        close $isotovideo;
-        return ($testpid, $child);
-    }
-
-    die "cannot fork: $!" unless defined $testpid;
-    close $child;
+    $process = process(sub {
+            close $child;
+            $SIG{TERM} = \&handle_sigterm;
+            $SIG{INT}  = 'DEFAULT';
+            $SIG{HUP}  = 'DEFAULT';
+            $SIG{CHLD} = 'DEFAULT';
 
-    $SIG{TERM} = \&handle_sigterm;
-    $SIG{INT}  = 'DEFAULT';
-    $SIG{HUP}  = 'DEFAULT';
-    $SIG{CHLD} = 'DEFAULT';
+            cv::init;
+            require tinycv;
 
-    cv::init;
-    require tinycv;
+            $0 = "$0: autotest";
+            my $line = <$isotovideo>;
```

Comment:
Is it safe to read in this way from a/this socket? Although I see below that it has always been read like this. I guess this type of read is buffered, so there better not be any unbuffered IO on this socket elsewhere in the code.

Reply:
Well, nope if you ask me, but as you said it always been like this - as this will already shake things a lot, better do in a separate PR imho (hence why i said there is a lot of room for improvement :) )